[TAS-114] 멤버 탈퇴 API

S-J-Kim opened the pull request:
## PR 체크리스트
Merge 전에 아래 체크리스트가 모두 체크되었는지 확인해주세요 

- [ ] 마지막 commit전에 빌드하여 코드 스타일 점검하였나요?
- [ ] 최소 1명의 리뷰와 Approve를 받았나요?
- [x] 모든 Test 코드를 실행하여 PASS했나요?
- [ ] 포스트맨 API 명세를 작성하였나요?


## PR 타입
PR의 타입을 체크해주세요

<!-- Please check the one that applies to this PR using "x". -->

- [ ] 버그 픽스 - 이슈 링크 : 
- [x] 신규 기능 개발
- [ ] 기능 수정
- [ ] 테스트 코드 추가
- [ ] 코드 스타일 업데이트 (formatting, 변수명 수정 등)
- [ ] 리팩토링 (기능 수정은 없고 코드 재사용성을 위한 메서드 분리 등)
- [ ] 설정 파일 수정
- [ ] 기타


## 코드 변경 이유

- 이슈 링크 : https://far-coconut-eec.notion.site/084472cb15f646d2a183dd0f3ffa82bb?pvs=4



## 변경된 사항

-  멤버 탈퇴 처리 API 개발
- 

## 테스트 방법

-
-

## 리뷰어 집중 사항

- 
- 


## 기타 전달 사항

- 
-

S-J-Kim committed:
feat: 멤버 탈퇴 API

gyun-ky commented on src/main/java/com/LetMeDoWith/LetMeDoWith/application/member/service/MemberService.java:
```diff
@@ -112,12 +120,31 @@ public void createMemberTermAgree(boolean isTermsAgree, boolean isPrivacyAgree,
+     * @return 탈퇴요청 성공 여부
+     */
+    public void withdrawMember(Long memberId) {
+        Optional<Member> optionalMember = memberRepository.getMember(memberId, MemberStatus.NORMAL);
```
orElseThrow로 Excpetion 처리하는게 코드상 깔쌈할거 같습니다

gyun-ky commented on src/main/java/com/LetMeDoWith/LetMeDoWith/domain/member/Member.java:
```diff
@@ -146,9 +149,15 @@ public Member updatePersonalInfoAfterCompleteSignUp(MemberPersonalInfoVO persona
     }
-
+    
     public void updateToNormalStatus() {
```
안쓰는 메서드인거 같습니다

S-J-Kim replied:
<img width="618" alt="image" src="https://github.com/user-attachments/assets/18e4e5e8-c36b-4522-b28a-3002c768cb86">

기록상으로는 이렇게 나오긴 합니다만,,, 필요 없어 보이니 지우도록 하겠습니다.

gyun-ky replied:
아 약간 도메인 멤버 메서드 예시로 제가 넣어놓은거 같네여 네네 얼추 DDD 이해하셨으니 지워주셔요

gyun-ky commented on src/main/java/com/LetMeDoWith/LetMeDoWith/infrastructure/member/repository/MemberRepositoryImpl.java:
```diff
@@ -1,69 +1,70 @@
+    }
+    
+    @Override
+    public void saveAgreement(Member member, MemberAgreementCommand command) {
```
해당 부분 제일 최신 PR에 코멘트 드렸습니다

`Repository에 너무 의존적아닌가`

S-J-Kim replied:
비즈니스 로직에도 오류가 있어 (save인데 무조건 init 객체를 저장) 수정하면서 겸사겸사 고쳐보겠습니다.

S-J-Kim committed:
fix: 리뷰 대응
- Repository에 의존적인 비즈니스 로직 Service layer로 이동
- Optional 객체 불러오는 부분 orElseThrow 사용으로 개선

S-J-Kim committed:
Resolve Merge Conflict
